src/Doctrine/Orm/State/CollectionProvider.php

```diff
@@ -59,7 +59,11 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
         $queryBuilder = $repository->createQueryBuilder('o');
         $queryNameGenerator = new QueryNameGenerator();
 
-        $this->handleLinks($queryBuilder, $uriVariables, $queryNameGenerator, $context, $entityClass, $operation);
+        if ($options instanceof Options && \is_callable($handleLinks = $options->getHandleLinks())) {
```

Comment:
This part should be looping through $options->gethandleLinks() shouldn't it? Do you think supporting a mix of custom and "default" Links is doable?

Reply:
no `handleLinks` is a callable, and I'll add a `service` in the future (requires a bit more work with a service locator etc.)